assets/js/modules/analytics/datastore/setup-flow.test.js

```diff
@@ -0,0 +1,327 @@
+						name: 'properties/1000',
+						createTime: '2014-10-02T15:01:23Z',
+						updateTime: '2014-10-02T15:01:23Z',
+						parent: 'accounts/100',
+						displayName: 'Test GA4 Property',
+						industryCategory: 'TECHNOLOGY',
+						timeZone: 'America/Los_Angeles',
+						currencyCode: 'USD',
+						deleted: false,
+					},
+				],
+				{ accountID: 'another-different-id' },
+				);
+
+				//  Receive empty properties list to prevent unexpected fetch by resolver.
+				registry.dispatch( MODULES_ANALYTICS_4 ).receiveGetProperties(
+					[],
+					{ accountID },
+				);
+				registry.dispatch( MODULES_ANALYTICS_4 ).receiveGetWebDataStreams(
+					[
+						{
+							_id: '2000',
+							_propertyID: '1000',
+							name: 'properties/1000/webDataStreams/2000',
+							// eslint-disable-next-line sitekit/acronym-case
+							measurementId: '1A2BCD345E',
+							// eslint-disable-next-line sitekit/acronym-case
+							firebaseAppId: '',
+							createTime: '2014-10-02T15:01:23Z',
+							updateTime: '2014-10-02T15:01:23Z',
+							defaultUri: 'http://example.com',
+							displayName: 'Test GA4 WebDataStream',
+						},
+					],
+					{ propertyID: 'foobar' }
+				);
+
+				expect( registry.select( MODULES_ANALYTICS_4 ).isAdminAPIWorking() ).toBe( true );
+
+				expect( registry.select( MODULES_ANALYTICS ).getSetupFlowMode() ).toBe( 'ua' );
+			} );
+
+			it( 'should return undefined if selected account returns an empty array from UA getProperties selector', () => {
+				registry = createTestRegistry();
+				// Receive empty settings to prevent unexpected fetch by resolver.
+				registry.dispatch( MODULES_ANALYTICS ).receiveGetSettings( {} );
+				registry.dispatch( MODULES_ANALYTICS ).setAccountID( accountID );
+
+				registry.dispatch( MODULES_ANALYTICS_4 ).receiveGetWebDataStreams(
+					[
+						{
+							_id: '2000',
+							_propertyID: '1000',
+							name: 'properties/1000/webDataStreams/2000',
+							// eslint-disable-next-line sitekit/acronym-case
+							measurementId: '1A2BCD345E',
+							// eslint-disable-next-line sitekit/acronym-case
+							firebaseAppId: '',
+							createTime: '2014-10-02T15:01:23Z',
+							updateTime: '2014-10-02T15:01:23Z',
+							defaultUri: 'http://example.com',
+							displayName: 'Test GA4 WebDataStream',
+						},
+					],
+					{ propertyID: 'foobar' }
+				);
+
+				// Need to dispatch empty UA properties so has loaded
+				registry.dispatch( MODULES_ANALYTICS ).receiveGetProperties(
+					[],
+					{ accountID },
+				);
+
+				fetchMock.get(
+					/^\/google-site-kit\/v1\/modules\/analytics-4\/data\/properties/,
+					{
+						body: [],
+						status: 200,
+					}
+				);
+
+				expect( registry.select( MODULES_ANALYTICS_4 ).getProperties( accountID ) ).toBe( undefined );
+
+				expect( registry.select( MODULES_ANALYTICS ).getSetupFlowMode() ).toBe( undefined );
+			} );
+
+			it( 'should return "ga4" if selected account returns an empty array from UA getProperties selector', () => {
```

Comment:
The great thing about the weekend @tofumatt  is that when Monday comes I can code review my own code!   :joy:  :sob: 

Note this test description and the previous one, they have the same input!!

```
it( 'should return undefined if selected account returns an empty array from UA getProperties selector', () => {


it( 'should return "ga4" if selected account returns an empty array from UA getProperties selector', () => {
```


Note the same _"if selected account returns an empty array from UA getProperties selector"_  :see_no_evil: 

That's an oversight!

I'm concerned that the IB and AC don't mention `undefined`, but our code does.  So it's going to be hard to QA

Reply:
Huh, whoops, that's right.

Returning `undefined` should only happen if a selector returns `undefined`—eg if things are in a "loading"` state.

If either `getProperties` are empty then we return the opposite "mode", which is correct. So I think everything here is covered.

`undefined` is assumed to be a loading state in our code, so it's fine that it wasn't mentioned, but you can note it in the QA brief if you like 🙂